hoss/coordinate_utilities.py

```diff
@@ -341,54 +326,75 @@ def create_dimension_arrays_from_coordinates(
+
+    y_dim = get_1d_dim_array_data_from_dimvalues(
+        row_dim_values, np.transpose(row_indices)[0], row_size
+    )
+
+    x_dim = get_1d_dim_array_data_from_dimvalues(
+        col_dim_values, np.transpose(col_indices)[1], col_size
     )
 
     projected_y, projected_x = tuple(projected_dimension_names)
-    return {projected_y: y_dim, projected_x: x_dim}
 
+    if dim_order_is_y_x:
+        return {projected_y: y_dim, projected_x: x_dim}
+    return {projected_x: x_dim, projected_y: y_dim}
 
-def get_dimension_array_from_geo_points(
-    lat_arr: np.ndarray,
-    lon_arr: np.ndarray,
-    crs: CRS,
-    dimension_indices: list,
-    dimension_size: int,
-    is_row=True,
-) -> np.ndarray:
-    """This function uses the list of lat/lon points corresponding
-    to a list of array indices and reprojects it with the CRS
-    provided and scales the x/y values to a dimension array with the dimension
-    size provided
 
+def get_dimension_order_and_dim_values(
+    lat_array_points: np.ndarray,
+    lon_array_points: np.ndarray,
+    dimension_indices: list,
+    crs: CRS,
+    is_row: bool,
+) -> bool:
+    """Determines the order of dimensions based on whether the
+    projected y or projected_x values are varying across row or column
+    Also returns the varying projected dimension values
     """
-    if is_row:
-        index_for_dimension = 0
-        x_or_y_index = 1
-    else:
-        index_for_dimension = 1
-        x_or_y_index = 0
+    # if lat/lon array is 2D and variables are also 2D
+    lat_arr_values = [lat_array_points[i][j] for i, j in dimension_indices]
+    lon_arr_values = [lon_array_points[i][j] for i, j in dimension_indices]
 
-    geo_grid_points = [
-        (lon_arr[row, col], lat_arr[row, col]) for row, col in dimension_indices
-    ]
-    x_y_values = get_x_y_values_from_geographic_points(geo_grid_points, crs)
-    indices_for_dimension = np.transpose(dimension_indices)[index_for_dimension]
-    dimension_values = np.transpose(x_y_values)[x_or_y_index]
-    return get_1d_dim_array_data_from_dimvalues(
-        dimension_values, indices_for_dimension, dimension_size
+    from_geo_transformer = Transformer.from_crs(4326, crs)
+    x_values, y_values = (  # pylint: disable=unpacking-non-sequence
+        from_geo_transformer.transform(lat_arr_values, lon_arr_values)
     )
+    y_variance = np.abs(np.diff(y_values))
+    x_variance = np.abs(np.diff(x_values))
+
+    # if it is row and projected_y is varying more than projected_x
+    # y_x order is true
+    if is_row is True:
+        if y_variance > x_variance:
+            return True, y_values
+        if x_variance > y_variance:
+            return False, x_values
+
+    # if it is col and projected_y is changing more than
+    # projected_x. it is x_y_order which means y_x order is false
+    if is_row is False:
+        if y_variance > x_variance:
+            return False, y_values
+        if x_variance > y_variance:
+            return True, x_values
```

Comment:
A few comments on this logic:

1) You should use `else` instead of just saying `if <something>` and then `if <the opposite of that something>`. For example, your outer conditions are `if is_row is True` and then `if is_row is False`. Just have `if is_row:` and then `else`. This makes is clearer that the conditions are mutually exclusive. (Also, you don't need to say `if is_row is True`, you can just directly say `if is_row`, because it is a boolean.)
2) I wonder if the outer `if is_row is True` and `if is_row is False` is unnecessary, and you could just collapse both of the two conditions down into a single block of code. Instead the first return value could just by tied to the value of `is_row`. The first sub-condition in both blocks returns a boolean that matches the value of `is_row`, and the second sub-condition in both blocks returns the opposite boolean value to `is_row`. The snippet below is quickly cobbled-together code, but isn't it equivalent to this entire set of conditions? (L386 to L398 at the time of the review comment being made)

```
if y_variance > x_variance:
    return is_row, y_values
else:
    return not is_row, x_values
```

Reply:
updated and tested - https://github.com/nasa/harmony-opendap-subsetter/pull/19/commits/e922df1288b04815191390566370a7281876ca54

Reply:
> if both were not varying - then it is an exception

Good point - I had missed that extra condition. But I think that kind of reinforces the point.

By using `if`/`elif`/`else` instead of distinct `if` blocks, you are more clearly indicating that the conditions are related and mutually exclusive, which is valuable. It also is marginally more efficient as, if you satisfy the first condition, the code won't check subsequent conditions. Keeping the `InvalidCoordinateData` out of any condition block altogether makes it even more distinct from the rest (meaning that, in my haste, I totally missed that it was a separate branch for the code).

> Also if there is a return statement I get warning that I should not use the else..

The problem here is a `pylint` thing, right (possibly a no-else-return or no-else-raise)? I think the other formatting tools we have via `pre-commit` (`black` and `ruff`) are fine with this, so I'd be inclined to ignore `pylint` in this case. (We could even update the [pylint command](https://github.com/nasa/harmony-opendap-subsetter/blob/DAS-2270/tests/run_tests.sh#L38) to disable this specific warning)

The alternative that would satisfy `pylint` would be something like:

```
if y_variance > x_variance:
    is_y_x_order = is_row
    dimension_values = y_values
elif x_variance > y_variance:
    is_y_x_order = not is_row
    dimension_values = x_values
else:
    raise InvalidCoordinateData(...)  # this might still trigger a no-else-raise R1720 warning.

return is_y_x_order, dimension_values
```

Another point here: I'm not sure the exception message is spot on. I think you get there if `y_variance == x_variance`. x and y both being constant is one way to do that (when they are both equal, and the variance for both is zero). There's also the case where the row and column variance are the same non-zero amount. That implies there are two different exceptions to handle here:

* `y_variance == x_variance and y_variance == 0` - neither the row nor the column are varying.
* `y_variance == x_variance and y_variance !=0` - both the row and the column are varying, and by the same amount.

I think that means you could either:

* Tweak the exception message to state that x and y are varying by the same amount (instead of saying they are constant).
* Separating out those two exceptions and having more specific exception messages for each.